src/System.Text.Utf8/src/System/Text/Utf8/Utf8CodeUnit.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace System.Text.Utf8
+{
+    public struct Utf8CodeUnit : IEquatable<Utf8CodeUnit>
```

Comment:
Why do we need this to be public? Why do we need a type at all? UTF8 code unit is byte, isn't it?